[lunasvg] Add usage

JonLiu1993 opened the pull request:
Fixes #30186

Lunasvg does not export anything after installation, I add the usage.
Tested the usage successfully by `lunasvg:x64-windows`:
````
lunasvg provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(unofficial-lunasvg CONFIG REQUIRED)
    target_link_libraries(main PRIVATE unofficial::lunasvg::lunasvg)
````

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [] ~~SHA512s are updated for each updated download~~
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

JonLiu1993 committed:
[lunasvg] Add usage

JonLiu1993 committed:
update version

JonLiu1993 committed:
update cmake

JonLiu1993 committed:
update version

JonLiu1993 committed:
update patch

JonLiu1993 committed:
update version

JonLiu1993 committed:
optimize portfile.cmake

JonLiu1993 committed:
update version

JonLiu1993 committed:
update cmake_function

JonLiu1993 committed:
update version

dg0yt commented:
Does this config help with any usage requirements beyond what documenting `find_path` and `find_library` in `usage` could do?
Adding unofficial config everywhere (and not upstreaming it) effectively still creates vcpkg lock-in.

JonLiu1993 commented:
> Does this config help with any usage requirements beyond what documenting `find_path` and `find_library` in `usage` could do? Adding unofficial config everywhere (and not upstreaming it) effectively still creates vcpkg lock-in.

@dg0yt, Thanks for your review, what exactly are you worthy of meeting any other requirements? I just checked locally whether the files generated under its `packages\` folder are correct, and tested its exported usage locally.

dg0yt commented:
I didn't claim it wouldn't be working.
This change is adding a patch, which is intrusive, to add an unofficial CMake config which doesn't exist outside the vcpgk world. I'm not convinced that this is more beneficial than useful.

With regard to transitive usage requirements, I found only the following: C++ linkage, libm for non-windows.

With regard to upstream, https://github.com/sammycage/lunasvg/pull/84 may need support.

Pospelove commented:
Thanks for taking a look at this port

Please consider also updating to the latest commit. The current version is outdated and lacks critical patches Like UB fix https://github.com/sammycage/lunasvg/pull/123 and much more

Thanks again

JonLiu1993 committed:
Merge branch 'master' of https://github.com/microsoft/vcpkg into dev/Jon/lunasvg

JonLiu1993 committed:
[lunasvg] update to latest commit fix issue

JonLiu1993 committed:
update version

JonLiu1993 committed:
Merge branch 'dev/Jon/lunasvg' of https://github.com/JonLiu1993/vcpkg into dev/Jon/lunasvg

JonLiu1993 committed:
update version

JonLiu1993 committed:
update version

JonLiu1993 committed:
update version

JonLiu1993 committed:
updatre version

vicroms commented on ports/lunasvg/vcpkg.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "lunasvg",
-  "version": "2.3.1",
+  "version": "2.3.1+20230331",
```
A couple of comments regarding the versioning of this port.

* Using the build tag (`+<build>`) is not a good idea. 
According to SemVer, build tags have no semantic meaning, e.g.: `2.3.1` is the same version as `2.3.1+20230331`; but in practice, vcpkg must treat them differently to avoid non-determinism when resolving versions. For this reason, I believe it is better to use the same base `version` and just increases the `port-version` field to communicate that there has been an "improvement". 
* Regarding specifically to `lunasvg`, the latest release tag is `2.3.5` (released 7 months ago), and the [version in their CMakeLists.txt reflects that](https://github.com/sammycage/lunasvg/blob/f924651b85cac47dbe15f51a4aa320461fc1d07b/CMakeLists.txt#L3). So, keeping `2.3.1` as the base version while using the latest commit is wrong.

```suggestion
  "version": "2.3.5",
```

JonLiu1993 replied:
@vicroms, Thanks for your suggesstion, version 2.3.5 does not include usage-related changes. Currently we have two solutions. The first is to use "version-date" to update to the latest commit, and the other is to update to the latest release version 2.3.5 and use patches to add usage, which one should we use?

vicroms replied:
I'd like to avoid changing the version scheme unless we know that upstream is not planning to release any new version tags. For that reason, I'd like to go with the approach of using patches to backport the CMake integration into the 2.3.5 release.

@sammycage are any more `lunasvg` releases planned in the near future?

JonLiu1993 committed:
update version

JonLiu1993 committed:
update version

aeris170 commented:
Sorry to rush things but, could we merge this?

Commit:
update lunasvg to 2.3.8

Commit:
update version

JonLiu1993 commented:
Since the addition of upstream usage caused [issue](https://github.com/sammycage/lunasvg/issues/131) for users, the latest upstream version restored the [changes](https://github.com/sammycage/lunasvg/pull/84) about usage in `CMakeLists`. I updated to the latest version and added a patch to provide unofficial usage.

JonLiu1993 commented:
I have commit a PR for Upstream https://github.com/sammycage/lunasvg/pull/132

Commit:
Fix conflicting file

Commit:
update version

JonLiu1993 commented:
I submitted a [PR](https://github.com/sammycage/lunasvg/pull/132) online, but there has been no response for a month and a half. I think we can merge this PR first for users to use.

JonLiu1993 commented on versions/l-/lunasvg.json:
```diff
@@ -1,9 +1,19 @@
       "git-tree": "064296e0fbadcd3113e5ee2a65202614c9125acc",
       "version": "2.3.8",
       "port-version": 0
+    },
+    {
+      "git-tree": "bf20380ca537151f7d1e02a6ca5b19c302db18c7",
+      "version": "2.3.1",
+      "port-version": 0
```
PR #32510 Deleted the historical "git-tree" when updating lunasvg, I restore it here.

BillyONeal replied:
Oops 😳. Thanks!

sakiodre replied:
My mistake, thanks for the fix!

BillyONeal commented:
Thanks!

aeris170 commented:
Thanks for merging fast!
